arangod/RocksDBEngine/RocksDBCollectionMeta.cpp

```diff
@@ -131,7 +132,7 @@ rocksdb::SequenceNumber RocksDBCollectionMeta::committableSeq() const {
     auto it = _blockersBySeq.begin();
     return it->first;
   }
-  return std::numeric_limits<rocksdb::SequenceNumber>::max();
+  return rocksutils::globalRocksDB()->GetLatestSequenceNumber();
```

Comment:
why is it necessary to access the latest sequence number here ?

Reply:
isn't there logic somewhere that filters out sequences larger than the latest one ? 
I am asking because this call is probably going to be expensive if we have 10000+ collections

Reply:
So the call to `committableSeq` here: https://github.com/arangodb/arangodb/blob/f36b4d6979ec7eebd81ee76391d8502893153cce/arangod/RocksDBEngine/RocksDBCollectionMeta.cpp#L202

This gets passed into `applyAdjustments`, which may do nothing and just spit it back out unmodified. If it doesn't do anywork, then the `std::min` will just spit it back out again here: https://github.com/arangodb/arangodb/blob/f36b4d6979ec7eebd81ee76391d8502893153cce/arangod/RocksDBEngine/RocksDBCollectionMeta.cpp#L207

It can then be passed into the index estimator here: https://github.com/arangodb/arangodb/blob/f36b4d6979ec7eebd81ee76391d8502893153cce/arangod/RocksDBEngine/RocksDBCollectionMeta.cpp#L276

My fix to avoid the 0 can then preserve this value through `applyUpdates`: https://github.com/arangodb/arangodb/blob/f36b4d6979ec7eebd81ee76391d8502893153cce/arangod/RocksDBEngine/RocksDBCuckooIndexEstimator.h#L606

Which can then fail this assertion: https://github.com/arangodb/arangodb/blob/f36b4d6979ec7eebd81ee76391d8502893153cce/arangod/RocksDBEngine/RocksDBCuckooIndexEstimator.h#L255

I agree that it would be better to have some better way to call this once and pass the saved value through, but that's a bit trickier to get the logic right. In any case, I don't think it's a good idea to have `committableSeq` return `::max()`. It should rather return the current sequence number, which it could possibly take as an input. Maybe that's the right way to do it?

Reply:
@graetzer Okay, I think I addressed this now by adding an input parameter to `committableSeq` to take the current sequence number (or really just a default, but it should in theory be essentially the current value). I do recall my original implementation had this input parameter as well, so it must have just gotten lost somewhere along the way.